### DISCUSSION TITLE
v2.1 - Mini player

### DISCUSSION THREAD

**Discussion by busybox11** [busybox11/NowPlaying-for-Spotify#79] [Announcements]
## What's new since v2.0.1?
- Mini player creation and generation page
  - You can use the mini player without authenticating yourself from the browser you'll be using.
    It will contain a usage token. Though, make sure not to share this to anyone.
  - This can be used, for instance, with OBS, as a "now playing" indicator for streams.
    Just copy and paste the generated URL into a browser source, set the right dimensions, and you'll be good to go.

![Usage example with OBS](https://github.com/busybox11/NowPlaying-for-Spotify/assets/29630035/9101845c-6bd3-4586-be06-fb47a7a3f1fa)
*Usage example with OBS*

<hr /><em>This discussion was created from the release <a href='https://github.com/busybox11/NowPlaying-for-Spotify/releases/tag/v2.1'>v2.1 - Mini player</a>.</em>

**Comment by ArchVaz** [comment]
Hello, Is there a way to use this mini player on a website without giving access to the token? Sorry I'm pretty new to this.

**Reply by busybox11** [reply]
Hello!
There is indeed a way to embed this on a website, albeit pretty tricky. You'd have to setup a proxy that would handle all the requests to NowPlaying's server, since they contain your access token.
Unfortunately I do not have anything ready to use to give you. If you are going in this route, feel free to reach out and I'll be happy to help!

I am working on a server-side implementation that would permit the use of the miniplayer safely using NowPlaying's server, but unfortunately this requires a lot of work and time that I don't have right now. It may not come any time soon.

<br>

In the meantime, I do have another solution, but it unfortunately requires you to host it yourself and it is not as straight-forward as clicking a button to use.

Two of my other projects, https://github.com/busybox11/spotify-websocket-client and https://github.com/busybox11/MyPlaying, allow you to have your own Spotify now playing server with its own frontend that you then can embed anywhere safely. You can see it in action on my personal portfolio (https://portfolio.busybox.colabois.fr/).

![image](https://github.com/busybox11/NowPlaying-for-Spotify/assets/29630035/cefe3c66-8cd7-497d-8925-9925445ccc9a)

There isn't barely any documentation and it's not easy to install. It's doable though - same goes for this, if you are going to use this solution, feel free to reach out and I'll be glad to help you figure it out and tell you how to do it.

<br>

Thanks for checking out my project, I hope my reply is understandable lol.

**Reply by ArchVaz** [reply]
Thank you so much for replying, I would like to try it but I have no idea what to do :(, I can host it as I'm running a dedicated pc 24/7 to run my radio station but if I need to code or something I'm just completely lost.

I will take so much time for your hands trying to help me with this.

**Reply by busybox11** [reply]
Hi, apologies for the late reply.

If you're already using a computer to host something 24/7, you could use it as a server then.
You don't have to code anything. You do have to install and setup the projects though, but I could absolutely help you with that.

Don't worry about it. I'm happy to help :) If you'd like to host the projects on your computer, feel free to send me a DM and I'll guide you through it.

I may have to do some changes on them because they're not really documented and I didn't really expect anyone besides me to use them. But this will be a nice addition no matter what!

**Reply by ArchVaz** [reply]
Hey no worries I know you are busy with life, and that would be great if you can help me out a little on this.

**Reply by busybox11** [reply]
Hey, it's ok :) Don't hesitate to send me a DM!
I might not be able to reply right away but I'll definitely be happy to help you out on this.

**Reply by ArchVaz** [reply]
I'm not really sure how to send DM here I could not find a way to do it.

**Reply by busybox11** [reply]
Oh there isn't any DMs here. You can find my socials on my GitHub profile page
